core/db_classes/EE_Base_Class.class.php

```diff
@@ -160,112 +209,90 @@ protected function __construct($fieldValues = [], $by_db = false, $timezone = ''
                 );
             }
         }
-        // remember what values were passed to this constructor
-        $this->_props_n_values_provided_in_constructor = $fieldValues;
-        // remember in entity mapper
-        if (! $by_db && $this->_model->has_primary_key_field() && $this->ID()) {
-            $this->_model->add_to_entity_map($this);
-        }
-        // setup all the relations
-        foreach ($this->_model->relation_settings() as $relation_name => $relation_obj) {
-            if ($relation_obj instanceof EE_Belongs_To_Relation) {
-                $this->_model_relations[ $relation_name ] = null;
-            } else {
-                $this->_model_relations[ $relation_name ] = [];
-            }
-        }
-        /**
-         * Action done at the end of each model object construction
-         *
-         * @param EE_Base_Class $this the model object just created
-         */
-        do_action('AHEE__EE_Base_Class__construct__finished', $this);
-    }
-
-
-    /**
-     * Gets whether or not this model object is allowed to persist/be saved to the database.
-     *
-     * @return boolean
-     */
-    public function allow_persist()
-    {
-        return $this->_allow_persist;
     }
 
 
     /**
-     * Sets whether or not this model object should be allowed to be saved to the DB.
-     * Normally once this is set to FALSE you wouldn't set it back to TRUE, unless
-     * you got new information that somehow made you change your mind.
+     * for getting a model while instantiated.
      *
-     * @param boolean $allow_persist
-     * @return boolean
+     * @param string|null $timezone
+     * @return EEM_Base | EEM_CPT_Base
+     * @throws EE_Error
+     * @throws ReflectionException
      */
-    public function set_allow_persist($allow_persist)
+    public function get_model($timezone = '')
     {
-        return $this->_allow_persist = $allow_persist;
+        if (! $this->_model) {
+            $timezone = $timezone ? $timezone : $this->_timezone;
+            $modelName    = self::_get_model_classname(get_class($this));
+            $this->_model = self::_get_model_instance_with_name($modelName, $timezone);
+            $this->set_timezone($timezone);
+        }
+        return $this->_model;
     }
 
 
     /**
-     * Gets the field's original value when this object was constructed during this request.
-     * This can be helpful when determining if a model object has changed or not
+     * Overrides parent because parent expects old models.
+     * This also doesn't do any validation, and won't work for serialized arrays
      *
      * @param string $field_name
-     * @return mixed|null
+     * @param mixed  $field_value_from_db
      * @throws InvalidArgumentException
      * @throws InvalidInterfaceException
      * @throws InvalidDataTypeException
      * @throws EE_Error
      */
-    public function get_original($field_name)
+    public function set_from_db($field_name, $field_value_from_db)
```

Comment:
YIKES!!!
this diff is NOT an accurate representation of the changes that were made.

Sometimes my IDE reorders methods, but I can't figure out how the configuration for that works... changes don't seem to do what their documentation says :\

I think I'll go through this file and return all of the methods to their original locations so that the diff isn't so mangled.

Reply:
Right, I understand that the diff is mangled here, but this one seemed legit 😬

Reply:
???
not sure what you mean... i think the change might be a bit confusing... so I'll explain (but my apologies if i totally miss your point and explain something you already know).

- the class constructor had a parameter named `$bydb` 
- toggling `$bydb` to true would cause the constructor to call `set_from_db()` instead of `set()` when initializing the field data.
- i renamed the `$bydb` parameter to match the `set_from_db()` method because I think that is more clear (plus I hate when a "previous dev" would constantly switch up var/param/func names and much prefer things to be consistent)
- `set_from_db()` was an existing method and was not renamed, only the `$bydb` parameter was renamed

Reply:
ahh kk gotcha now 👍🏻